modules/api/src/main/java/org/apache/ignite/compute/IgniteCompute.java

```diff
@@ -48,9 +49,29 @@ public interface IgniteCompute {
      * @param arg Argument of the job.
      * @return Job execution object.
      */
+    default <T, R> JobExecution<R> submit(
+            JobTarget target,
+            JobDescriptor<T, R> descriptor,
+            @Nullable T arg
+    ) {
+        return submit(target, descriptor, null, arg);
+    }
+
+    /**
+     * Submits a {@link ComputeJob} of the given class for an execution on a single node from a set of candidate nodes.
+     *
+     * @param <T> Job argument (T)ype.
+     * @param <R> Job (R)esult type.
+     * @param target Execution target.
+     * @param descriptor Job descriptor.
+     * @param cancellationToken Cancellation token or {@code null}.
+     * @param arg Argument of the job.
+     * @return Job execution object.
+     */
     <T, R> JobExecution<R> submit(
             JobTarget target,
             JobDescriptor<T, R> descriptor,
+            @Nullable CancellationToken cancellationToken,
             @Nullable T arg
     );
```

Comment:
all `submit*` methods should be left intact, since they return `JobExecution` which is handle itself containing `cancel` method

Reply:
done